### PR TITLE
Pass authenticated user to cylc-flow

### PR DESCRIPTION
**Pull request by datamel**
This is necessary for logging which authenticated user executes the command on the UI.
Sibling pr: https://github.com/cylc/cylc-flow/pull/4522
These changes close https://github.com/cylc/cylc-uiserver/issues/224 and https://github.com/cylc/cylc-flow/issues/4294

- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Does not need tests - can not test this currently due to no multi user testing in the automated test battery.
- [x] No change log entry required -invisible to users.
- [x] No documentation update required.
- [x] No dependency changes.

### CODE REVIEW

**Review by wxtim — APPROVED** [review]

**Review by wxtim — COMMENTED** [review]
- [x] Code makes sense
- [x] Demoed
- [ ] Passing tests

**Comment by datamel** [comment]
Note test failure expected as it requires sibling cylc-flow to be checked out.

> * [x]  Code makes sense
> * [x]  Demoed
> * [ ]  Passing tests

Ah, tests will only pass if checked out with sibling on cylc-flow.

**Comment by wxtim** [comment]
I can confirm that tests pass if Sibling PR is checked out. I will leave you to merge this and the sibling at the same time so as not to break master.

**Review by wxtim — APPROVED** [review]

File: cylc/uiserver/resolvers.py [hunk 1]
```diff
@@ -189,7 +189,10 @@ def __init__(self, data, log, **kwargs):
     # Mutations
     async def mutator(self, info, *m_args):
         """Mutate workflow."""
-        _, w_args, _ = m_args
+        req_meta = {}
+        _, w_args, _, _ = m_args
```
Comment (MetRonnie) [inline cylc/uiserver/resolvers.py:193]: Tuple unpacking has gained an extra member by mistake?
```suggestion
        _, w_args, _ = m_args
```

Comment (datamel) [reply cylc/uiserver/resolvers.py:198]: I know the test is failing here due to this but you should find that if you have the sibling checked out on cylc-flow then the test will pass (with the correct unpacking). The unpacking accounts for the meta.

File: cylc/uiserver/resolvers.py [hunk 2]
```diff
@@ -224,29 +227,3 @@ async def service(self, info, *m_args):
             self.workflows_mgr,
             log=self.log
         )
-
-    async def nodes_mutator(self, info, *m_args):
```
Comment (datamel) [inline cylc/uiserver/resolvers.py:228]: @dwsutherland, any chance you could double check that this is unused code. I couldn't get any breakpoints stopping here, can't find any code calls. Both myself and @oliver-sanders think it is not used and should be removed? Also, removed its cylc-flow partner in https://github.com/cylc/cylc-flow/pull/4522/files.

File: cylc/uiserver/resolvers.py (hunk 1 above)
Comment (MetRonnie) [reply cylc/uiserver/resolvers.py:198]: Oh, it's also done in https://github.com/cylc/cylc-flow/pull/4522. It's generally a good idea to have the same people review sibling PRs

Comment (wxtim) [reply cylc/uiserver/resolvers.py:198]: I'm tempted to ask GH for a multi-repo PR feature - surely we aren't the only people doing this sort of thing....

Comment (datamel) [reply cylc/uiserver/resolvers.py:198]: It seems strange GH actions can't check out different branches. Maybe something to look into when time permits.

Comment (oliver-sanders) [reply cylc/uiserver/resolvers.py:198]: Kinda doable, we could put a special comment in the PR description that tells GHA what base branch to use, however, the failing tests are also a good indicator not to merge until the dependent branch is in.

**Review by oliver-sanders — APPROVED** [review]
(tests pass locally with the cylc-flow branch checked out)

**Review by MetRonnie — APPROVED** [review]

File: cylc/uiserver/resolvers.py (hunk 2 above)
Comment (dwsutherland) [reply cylc/uiserver/resolvers.py:228]: It was in use, until:
https://github.com/cylc/cylc-flow/pull/3469
specifically:
https://github.com/cylc/cylc-flow/pull/3469/files#diff-97f4d62be41218e2d70a203edfca6d44d7c0046b6908cefa8e3af7c02cd31920

And the idea behind it was that mutations being multi-workflow could be given node ids from different workflows, and then these ids would be parsed and mutations sent to the appropriate workflow..

The fact that we moved away from this, means we are happy with not using this functionality... i.e. a node mutation will always be associated with only one workflow and/or the workflow parsing happens somewhere else.

**Pull request merged**